lib/modules/manager/kotlin-script/extract.ts

```diff
@@ -0,0 +1,38 @@
+import is from '@sindresorhus/is';
+import { regEx } from '../../../util/regex';
+import { MavenDatasource } from '../../datasource/maven';
+import type { PackageDependency, PackageFile } from '../types';
+
+const dependsOnRegex = regEx(
+  /@file\s*:\s*DependsOn\s*\(\s*(?<replaceString>"(?<groupId>.+):(?<artifactId>.+):(?<version>.+)")\s*\)/g
+);
+const repositoryRegex = regEx(
+  /@file\s*:\s*Repository\s*\(\s*"(?<repositoryName>.+)"\s*\)/g
+);
+
+export function extractPackageFile(fileContent: string): PackageFile | null {
+  const registryUrls: string[] = [...fileContent.matchAll(repositoryRegex)]
+    .map((match) => match.groups?.repositoryName)
+    .filter(is.string);
+
+  const matches = [...fileContent.matchAll(dependsOnRegex)];
+  const deps: PackageDependency[] = [];
+  for (const match of matches) {
+    const dep: PackageDependency = {
+      currentValue: match.groups?.version,
+      depName: `${match.groups?.groupId}:${match.groups?.artifactId}`,
+      replaceString: match.groups?.replaceString,
```

Comment:
Is it actually possible that `groups` is undefined? Because if it's undefined, we'd end up with a dep with undefined currentValue, depName and replaceString.

Reply:
No, in practice it's not possible - the question marks are there only because the typings say so. I added an extra check to be sure the groups are defined.

Reply:
Can the question marks be removed now that you added the extra if?

Reply:
Right, yes - the compiler now allows it, done.